clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -388,43 +391,39 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         // by markers.
         bool have_cond_branch_target = false;
         addr_t cond_branch_target = 0;
-        if (shard->prev_instr_.instr.addr != 0 /*first*/ &&
-            type_is_instr_direct_branch(shard->prev_instr_.instr.type) && has_encoding) {
-            if (shard->prev_instr_.instr.encoding_is_new)
+        if (type_is_instr_direct_branch(memref.instr.type) && expect_encoding) {
```

Comment:
Why did we remove the `first` check?

Reply:
Was doing this change on the current instr and check this check would have been unnecessary in that case but now that we went back to checking the previous instr for decoding check, I have added it back.